tempodb/encoding/vparquet3/schema.go

```diff
@@ -708,7 +726,32 @@ func parquetToProtoAttrs(parquetAttrs []Attribute, counter droppedAttrCounter, i
 	return protoAttrs
 }
 
-func parquetToProtoEvents(parquetEvents []Event) []*v1_trace.Span_Event {
+func parquetToLinks(parquetLinks []Link) []*v1_trace.Span_Link {
+	var protoLinks []*v1_trace.Span_Link
+
+	if len(parquetLinks) > 0 {
+		protoLinks = make([]*v1_trace.Span_Link, 0, len(parquetLinks))
+		for _, l := range parquetLinks {
+			protoLink := &v1_trace.Span_Link{
+				TraceId:                l.TraceID,
+				SpanId:                 l.SpanID,
+				TraceState:             l.TraceState,
+				DroppedAttributesCount: uint32(l.DroppedAttributesCount),
+				Attributes:             nil,
+			}
+
+			if len(l.Attrs) > 0 {
+				protoLink.Attributes = parquetToProtoAttrs(l.Attrs, &l, false)
+			}
+
+			protoLinks = append(protoLinks, protoLink)
+		}
+	}
+
+	return protoLinks
+}
+
+func parquetToProtoEvents(parquetEvents []Event, span Span) []*v1_trace.Span_Event {
```

Comment:
Span is a very large struct, it might be a good idea to pass it as a pointer `*Span` (or pass the span start time instead)